20231214-listUsers-api.md

```diff
@@ -0,0 +1,887 @@
+
+* You may see references to strings formatted as **`object#relation@user`** in this document. This is short-hand notation for representing an OpenFGA Relationship Tuple. For example, `group:eng#member@user:jon` or `document:1#viewer@group:eng#member` etc.. You may also see strings formatted as `objectType#relation`. This is short-hand notation for representing a specific relation defined on some object type. For example, `document#viewer` represents the viewer relationship defined on the document object type.
+
+* **Expansion** - refers to the process of iteratively traversing the graph of relationships that are realized through both an OpenFGA model AND the relationship tuples that exist in the store.
+
+* **Forward expansion** - refers to expansion of the OpenFGA relationship graph in a forward form. That is, starting at an object and relation, walk the graph of relationships in a directed way towards a user of some specific type by following paths that would lead through some target relation.
+
+* **Reverse expansion** - refers to expansion of the OpenFGA relationship grpah in a backwards form. That is, starting at a user of a specific type, walk the graph of relationships in a directed way towards an object of some specific type by following paths that would lead through some target relation.
+
+* **Concrete/terminal objects** - a concrete or terminal object refers to a singular or specific resource of a given object type and id. A concrete object cannot be expanded any further (see Expansion above). It can be conceptualized as a leaf-node in a graph. Concrete objects differ from usersets, because usersets refer to a collection of zero or more concrete objects. For example `user:jon` is a concrete object while `group:eng#member` is a userset which may expand to multiple concrete objects such as `user:jon`, `user:andres`, etc..
+
+* **Userset** - A userset refers to a set of zero or more concrete objects that are the subjects that a relationship expands to. See our Concepts page on [Users](https://openfga.dev/docs/concepts#what-is-a-user) for more info.
+
+
+## Motivation
+[motivation]: #motivation
+
+Developers using OpenFGA want to look up all of the users that have a particular relationship with an object. 
+
+### Use Cases
+
+- **UIs** - Display the users that a resource has been shared with. Ex: the "Share" dialog in Google Docs.
+- **Notifications** - Notify all users who relate to a specific object. Ex: send and email to editors of a document when it has been updated.
+
+### Expected Outcome
+
+Two new core APIs are added to the [OpenFGA API](https://github.com/openfga/api) and implementations of these new APIs in the [OpenFGA server](https://github.com/openfga/openfga).
+
+## What it is
+[what-it-is]: #what-it-is
+
+Given an `object`, `relation`, and one or more user/subject provided filters, return the concrete/terminal set of user objects matching at least one of the user filters that have that relationship with the object.
```

Comment:
i don't think `return the concrete/terminal set of user objects` is correct. By my [other](https://github.com/openfga/rfcs/pull/15/files#r1515256100) comment, I think this should day `return the concrete/terminal users, or sets of users, ...`